### PR TITLE
[hardware] sipeed slogic new hardware support

### PR DESCRIPTION
**Pull request by taorye**
As the title says, a new hardware want to be supported in mainline.
If there is something I do not prepared or missed, please give me some advice and I will as soon as possible to handle it.

### CODE REVIEW

**Comment by taorye** [comment]
`$9 for SLogic Lite8, $14.9 for SLogic Combo8(LA+4UART+CKLink+DAPLink)`

SLogic Lite8 (Logic Analyzer) 

channel: 8CH (MAX)
samplerate: 160Msps (MAX)
bandwidth: 320Mb/s (MAX)

supported configuration for continuously sampling
- 160M@2CH
- 80M@4CH
- 40M@8CH

input signal level: 0～3.6V (TTL)

**Review by taorye — COMMENTED** [review]
I have reviewed.

File: configure.ac [hunk 1]
```diff
@@ -356,6 +356,7 @@ SR_DRIVER([SCPI PPS], [scpi-pps])
 SR_DRIVER([serial DMM], [serial-dmm], [serial_comm])
 SR_DRIVER([serial LCR], [serial-lcr], [serial_comm])
 SR_DRIVER([Siglent SDS], [siglent-sds])
+SR_DRIVER([Sipeed Slogic Analyzer], [sipeed-slogic-analyzer])
```
Comment (MartinHerren) [inline configure.ac:359]: Not a libsigrok developer, but shouldn't a dependency to libusb be added here ?

Comment (MartinHerren) [reply configure.ac:359]: I can confirm, this is needed to prevent it from trying to build and fail if libusb is not available

**Comment by ANAU-designs** [comment]
Very interesting driver for this new very little logic analyzer (and more...)
Would be nice to also add the udev rules, update [contrib/60-libsigrok.rules]

`@@ -312,6 +312,9 @@ ATTRS{idVendor}=="10c4", ATTRS{idProduct}=="ea60", ENV{ID_SIGROK}="1"`
` # UNI-T UT-D09 multimeter cable (for various UNI-T and rebranded DMMs)`
` ATTRS{idVendor}=="10c4", ATTRS{idProduct}=="ea80", ENV{ID_SIGROK}="1"`
` `
`+# Sipeed SLogic Combo 8`
`+ATTRS{idVendor}=="359f", ATTRS{idProduct}=="0300", ENV{ID_SIGROK}="1"`
`+`
` # Sysclk LWLA1016`
` ATTRS{idVendor}=="2961", ATTRS{idProduct}=="6688", ENV{ID_SIGROK}="1"`

**Comment by MartinHerren** [comment]
> Very interesting driver for this new very little logic analyzer (and more...) Would be nice to also add the udev rules, update [contrib/60-libsigrok.rules]
> 
> `@@ -312,6 +312,9 @@ ATTRS{idVendor}=="10c4", ATTRS{idProduct}=="ea60", ENV{ID_SIGROK}="1"` ` # UNI-T UT-D09 multimeter cable (for various UNI-T and rebranded DMMs)` ` ATTRS{idVendor}=="10c4", ATTRS{idProduct}=="ea80", ENV{ID_SIGROK}="1"` ` ` `+# Sipeed SLogic Combo 8` `+ATTRS{idVendor}=="359f", ATTRS{idProduct}=="0300", ENV{ID_SIGROK}="1"` `+` ` # Sysclk LWLA1016` ` ATTRS{idVendor}=="2961", ATTRS{idProduct}=="6688", ENV{ID_SIGROK}="1"`

Yes, i already fixed that on my branch: https://github.com/sigrokproject/libsigrok/pull/212#issuecomment-1751844448
Changes
- Added udev rules
- Fixed channel number selection to get again 8 channels when selecting 40 MS/s or less
- Fixed build dependencies

I don't know if @taorye is still around and wants to update his PR or if we should open a new one.

There are still issues that the it regularly freezes and needs unplugging/replugging.

**Comment by MartinHerren** [comment]
Hej,

we'll see. If i have time to fix the usb issue and the original author didn't react, i'll probably open a new PR.

File: configure.ac (hunk 1 above)
Comment (taorye) [reply configure.ac:359]: You are right, libusb should be added.
Thanks :)

**Comment by taorye** [comment]
Thanks for your attention

> > Very interesting driver for this new very little logic analyzer (and more...) Would be nice to also add the udev rules, update [contrib/60-libsigrok.rules]
> > `@@ -312,6 +312,9 @@ ATTRS{idVendor}=="10c4", ATTRS{idProduct}=="ea60", ENV{ID_SIGROK}="1"` ` # UNI-T UT-D09 multimeter cable (for various UNI-T and rebranded DMMs)` ` ATTRS{idVendor}=="10c4", ATTRS{idProduct}=="ea80", ENV{ID_SIGROK}="1"` ` ` `+# Sipeed SLogic Combo 8` `+ATTRS{idVendor}=="359f", ATTRS{idProduct}=="0300", ENV{ID_SIGROK}="1"` `+` ` # Sysclk LWLA1016` ` ATTRS{idVendor}=="2961", ATTRS{idProduct}=="6688", ENV{ID_SIGROK}="1"`
> 
> Yes, i already fixed that on my branch: [master...MartinHerren:libsigrok:slogiclite8](https://github.com/sigrokproject/libsigrok/compare/master...MartinHerren:libsigrok:slogiclite8) Changes
> 
>     * Added udev rules
> 
>     * Fixed channel number selection to get again 8 channels when selecting 40 MS/s or less
> 
>     * Fixed build dependencies
> 
> 
> I don't know if @taorye is still around and wants to update his PR or if we should open a new one.
> 
> There are still issues that the it regularly freezes and needs unplugging/replugging.

I'll come back soon. There are more to be updated before next year.

**Comment by yjun123** [comment]
sample rate @ 1MHZ, 2MHZ, 4MHZ, 5MHZ

**Comment by yjun123** [comment]
Hello, gyus.

I have attempted to patch the mainline of libsigrok with MartinHerren's branch, specifically the [slogiclite8](https://github.com/MartinHerren/libsigrok/tree/slogiclite8) branch. I have tested it with the PulseView mainline.

I have encountered an issue where setting the sampling frequency to 5MHz or below can lead to data acquisition freeze. This issue is 100% reproducible.

In the case of a sampling rate greater than 5MHz, it works fine without any issues (?

my device is the Sipeed Slogic Combo 8.

**Comment by mkroman** [comment]
> So is it dead?

Certainly looks like it to me. There hasn't been a firmware release since 2023-09 for the SLogic Combo 8, and the provided PulseView AppImage hasn't been updated since 2023-08.. and I don't think there's another PR.

Looks a lot like vaporware tbh.. Hopefully development picks up again at some point. I don't understand why it wasn't at least made open source if there's not enough resources to work on it. :/

**Comment by g5pw** [comment]
Hm, I just noticed it actually uses the [M0S](https://wiki.sipeed.com/hardware/en/maixzero/m0s/m0s.html) module inside, apparently!

**Comment by MartinHerren** [comment]
Hej all,

Sorry for the late reply, didn't have time to use my Sipeed SLogic recently.

Juste rebased and rebuilt by branch (with @taorye's original support plus my fixes) on latest libsigrock's master branche.

Will try to use it during next week to try to reproduce @yjun123's issue and also my USB freezes.

I don't think it is a 'dead project' as they are still selling it. They haven't published new firmware and software as they might consider it works 'good enough' for them. Hopefully a stable version will end up officially in libsigrock.

I'm not even sure github is the correct place to open a PR for libsigrock as they never reacted to this PR. It is a read-only mirror but they say 'PR welcome'. For the moment i'll try to keep my branch alive and up to date with upstream to be able to continue to use my own device.

If it works stable enough and other people including upstream are interested i can open a new pull request.

My branch: https://github.com/MartinHerren/libsigrok/tree/slogiclite8, rebased on today's upstream:master.

**Comment by axel-h** [comment]
@MartinHerren since this PR is close due to @taorye deleting the branch, are you planning to open a new PR from your branch?

**Comment by 0xDRRB** [comment]
Sipeed [provides](https://wiki.sipeed.com/hardware/en/logic_analyzer/combo8/use_logic_function.html#Downloading-and-Running-the-Host-Computer) an [AppImage](https://dl.sipeed.com/shareURL/SLogic/SLogic_combo_8/4_application/PulseView) binary version of Pulseview with a libsigrok modified to support this hardware, without releasing the source code. Isn't that a violation of the GPL?

**Comment by g5pw** [comment]
A request was made here https://github.com/sipeed/sipeed_wiki/issues/698, I've actually just emailed support to request the source code.

**Comment by trabucayre** [comment]
@0xDRRB, @g5pw : after this request, email and [this post](https://x.com/SipeedIO/status/1833809669120315779), @sipeed has opensourced his repository [here](https://github.com/sipeed/sigrok_slogic).
Haven't compared the repo and this PR but I'm not sure there is more information to obtain...

**Comment by armandas** [comment]
Confusingly, there are two branches in [sipeed/sigrok_slogic ](https://github.com/sipeed/sigrok_slogic/branches/all) with very different implementations. Can anyone tell from the code which one would be better suited for merging into the mainline?

**Comment by ForsakenHarmony** [comment]
Seems like the repo is still getting updates

**Comment by taorye** [comment]
The process of integrating the latest developments into the main branch has officially commenced. Community members and contributors are invited to monitor the progress through the dedicated Pull Request [sigrokproject/libsigrok#262](https://github.com/sigrokproject/libsigrok/pull/262). This PR serves as a pivotal point for tracking updates, facilitating discussions, and ensuring transparency in the development cycle.

**Comment by MartinHerren** [comment]
Oh, great news ! Thanks !

Sorry didn't have any time for follow up recently, still have my SLogic 8 and ready to try it again !

**Pull request closed without merging**